### ISSUE TITLE
[FEATURE REQUEST] - Item Tags and Custom Fields

### ISSUE THREAD

**Issue by dpkonofa** [IgnisDa/ryot#578]
I would like to be able to specify custom fields for specific media types and have global and per-media tags that can be used within the various libraries and collections in RYOT.

As an example, I would like to be able to customize what types of fields are available for Movies. In my specific case, I own several physical movie releases that include standard releases, Steelbooks, Criterion Editions, and others. For some movies, I may own a Criterion Blu-Ray release but also a 4K UHD Steelbook, for example.

I would like to see some sort of way to edit the template for a type, like Movies (but not limited to it), where I can add fields that have some logic to them. If a movie is marked as "Owned" in RYOT (and only if it's marked as owned), I would like to have a <select> field that shows up that lets me select whether it's owned "digitally" or "physically". If I choose "physically", I would like to have options that ask what format and edition it is. Ideally, this wouldn't be hardcoded into RYOT so that it could be used to create and populate all kinds of fields for any type of tracker/collection.

The same might apply for music releases also. Since an album can have a CD, SACD, Blu-Ray, Limited Edition, Deluxe Edition, or other release (like the remaster of Dark Side of the Moon), it would be nice to have some way to track these features.

**Comment by IgnisDa** [comment]
Related to #73.

**Comment by IgnisDa** [comment]
Will look into this after #73 is implemented.

Will it work for you if there was a "Description" field while adding media to a collection?

**Comment by IgnisDa** [comment]
Makes sense, will think about how I can implement it.

**Comment by dpkonofa** [comment]
I'm happy to help if there's some way that I can.

There used to be a Mac app by Filemaker called Bento that did this really well. You set up item "templates" and then could create collections based on those templates. Editing a template edited all the items that used that template.

**Comment by IgnisDa** [comment]
I don't have plans to work on #73 in the near future, so these features will be much further down the line. I will start thinking about these things when I start to implement them.

For the time being, I am just collecting these feature requests so that I can get a general idea of what people want.

**Comment by IgnisDa** [comment]
@dpkonofa You mention Filemaker, but it is a Mac app, so I can not look into it. Are you aware of any other software that does this available on other platforms? I do not own Apple hardware.

**Comment by dpkonofa** [comment]
@IgnisDa - The default iOS contacts app and the Android Contacts app do something similar. You can add phone numbers, for example, but add as many as you want and change the label for each one - home, mobile, work, fax, etc.

It would just give the ability to add *any* field like that to a collection. So I could, for example, add a field called "Release Type" and set the options for that field to "DVD", "Blu-Ray", "4K Blu-Ray", "Digital", etc.

Another example of this is like Google Forms or Ninja Forms on Wordpress. They have pre-defined value/data types (single line text field, multi-line text field, number, dropdown, etc. and you can add any data type to a form, give it its own label, and then every form based on that template has that field.

**Comment by IgnisDa** [comment]
I will add a new tab in the preferences settings called "Tagging". Each tag will be allowed to have multiple attributes of string or number type:

```json
[
	{
		"name": "Inventory",
		"attributes": [
			{
				"name": "BoxNumber",
				"type": "number",
                                "optional": true
			},
			{
				"name": "RoomName",
				"type": "string"
			}
		]
	}
]
```

So then, when you associate a movie with the "Inventory" tag, Ryot will ask you to enter a "BoxNumber" and a "RoomName".

@dpkonofa What do you think about this?

**Comment by IgnisDa** [comment]
Released as part of [Ryot Pro](https://ryot.io).

**Comment by Treygec** [comment]
Hi, sorry if this is annoying but I just upgraded to pro and don't see this feature anywhere. Do I need to do something to surface it?

**Comment by IgnisDa** [comment]
When creating or editing a collection, you can  configure this. Once done,
you will get the option to enter data while adding an item to that
collection.
